ok next step @supriya-premkumar let's figure out why the following unit test is failing:

```
=== RUN   TestReorganizeTaints
    taints_test.go:593: [add new taint] expect to see operation tainted, but got: modified
--- FAIL: TestReorganizeTaints (0.00s)
```

Can you try reproduce the issue locally using:
```go test ./pkg/util/taints -run TestReorganizeTaints```

--- a/hack/verify-golangci-lint.sh
+++ b/hack/verify-golangci-lint.sh
@@ -49,4 +49,5 @@ golangci-lint run \
   --disable-all \
   -E deadcode \
   -E unused \
-  -E varcheck
+  -E varcheck \
+  -E ineffassign

--- a/pkg/apis/autoscaling/v1/conversion.go
+++ b/pkg/apis/autoscaling/v1/conversion.go
@@ -362,6 +362,7 @@ func Convert_autoscaling_HorizontalPodAutoscaler_To_v1_HorizontalPodAutoscaler(i
 		}
 		// copy before mutating
 		if !copiedAnnotations {
+			//nolint:ineffassign
 			copiedAnnotations = true
 			out.Annotations = autoscaling.DeepCopyStringMap(out.Annotations)
 		}

--- a/pkg/apis/autoscaling/v2beta1/conversion.go
+++ b/pkg/apis/autoscaling/v2beta1/conversion.go
@@ -299,6 +299,7 @@ func Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta1_HorizontalPodAutosca
 		}
 		// copy before mutating
 		if !copiedAnnotations {
+			//nolint:ineffassign
 			copiedAnnotations = true
 			out.Annotations = autoscaling.DeepCopyStringMap(out.Annotations)
 		}

--- a/pkg/controller/replicaset/replica_set_test.go
+++ b/pkg/controller/replicaset/replica_set_test.go
@@ -1125,7 +1125,11 @@ func TestDeleteControllerAndExpectations(t *testing.T) {
 	manager.deleteRS(rs)
 	manager.syncReplicaSet(GetKey(rs, t))
 
-	if _, exists, err = manager.expectations.GetExpectations(rsKey); exists {
+	_, exists, err = manager.expectations.GetExpectations(rsKey)
+	if err != nil {
+		t.Errorf("Failed to get controllee expectations: %v", err)
+	}
+	if exists {
 		t.Errorf("Found expectations, expected none since the ReplicaSet has been deleted.")
 	}
 

--- a/pkg/registry/core/service/storage/storage.go
+++ b/pkg/registry/core/service/storage/storage.go
@@ -70,8 +70,8 @@ func NewGenericREST(optsGetter generic.RESTOptionsGetter, serviceCIDR net.IPNet,
 
 	ipv4 := api.IPv4Protocol
 	ipv6 := api.IPv6Protocol
-	var primaryIPFamily *api.IPFamily = nil
-	var secondaryFamily *api.IPFamily = nil
+	var primaryIPFamily *api.IPFamily
+	var secondaryFamily *api.IPFamily
 	if netutil.IsIPv6CIDR(&serviceCIDR) {
 		primaryIPFamily = &ipv6
 		if hasSecondary {

--- a/pkg/util/taints/taints.go
+++ b/pkg/util/taints/taints.go
@@ -192,7 +192,7 @@ func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bo
 	allErrs := []error{}
 	var removed bool
 	for _, taintToRemove := range taintsToRemove {
-		removed = false
+		removed = false // nolint:ineffassign
 		if len(taintToRemove.Effect) > 0 {
 			*newTaints, removed = DeleteTaint(*newTaints, &taintToRemove)
 		} else {

--- a/pkg/volume/azuredd/azure_mounter.go
+++ b/pkg/volume/azuredd/azure_mounter.go
@@ -102,7 +102,6 @@ func (m *azureDiskMounter) SetUpAt(dir string, mounterArgs volume.MounterArgs) e
 			klog.Errorf("azureDisk - Unmount directory %s failed with %v", dir, err)
 			return err
 		}
-		mountPoint = true
 	}
 
 	if runtime.GOOS != "windows" {

--- a/test/images/agnhost/netexec/netexec.go
+++ b/test/images/agnhost/netexec/netexec.go
@@ -321,15 +321,12 @@ func dialHandler(w http.ResponseWriter, r *http.Request) {
 	var dialer func(string, net.Addr) (string, error)
 	switch strings.ToLower(protocol) {
 	case "", "http":
-		protocol = "http"
 		dialer = dialHTTP
 		addr, err = net.ResolveTCPAddr("tcp", hostPort)
 	case "udp":
-		protocol = "udp"
 		dialer = dialUDP
 		addr, err = net.ResolveUDPAddr("udp", hostPort)
 	case "sctp":
-		protocol = "sctp"
 		dialer = dialSCTP
 		addr, err = sctp.ResolveSCTPAddr("sctp", hostPort)
 	default:

--- a/test/integration/apiserver/apiserver_test.go
+++ b/test/integration/apiserver/apiserver_test.go
@@ -1285,11 +1285,6 @@ func TestAPICRDProtobuf(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			rv, _ := strconv.Atoi(obj.GetResourceVersion())
-			if rv < 1 {
-				rv = 1
-			}
-
 			w, err := client.Get().
 				Resource(resource).NamespaceIfScoped(obj.GetNamespace(), len(obj.GetNamespace()) > 0).Name(obj.GetName()).SubResource(tc.subresource).
 				SetHeader("Accept", tc.accept).

--- a/test/integration/deployment/deployment_test.go
+++ b/test/integration/deployment/deployment_test.go
@@ -224,12 +224,11 @@ func TestDeploymentSelectorImmutability(t *testing.T) {
 	}
 
 	// test to ensure apps/v1 selector is immutable
-	newSelectorLabels := map[string]string{"name_apps_v1beta1": "test_apps_v1beta1"}
 	deploymentAppsV1, err := c.AppsV1().Deployments(ns.Name).Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
 		t.Fatalf("failed to get apps/v1 deployment %s: %v", name, err)
 	}
-	newSelectorLabels = map[string]string{"name_apps_v1": "test_apps_v1"}
+	newSelectorLabels := map[string]string{"name_apps_v1": "test_apps_v1"}
 	deploymentAppsV1.Spec.Selector.MatchLabels = newSelectorLabels
 	deploymentAppsV1.Spec.Template.Labels = newSelectorLabels
 	_, err = c.AppsV1().Deployments(ns.Name).Update(context.TODO(), deploymentAppsV1, metav1.UpdateOptions{})

